feat: install `atlas` translations tool

OmarIthawi opened the pull request:
## Description

Add the [openedx-atlas](https://github.com/openedx/openedx-atlas) to the kernel.in so it's available in all environments including Atlas, Devstack and Native installation.

This is part 1 out of several other steps to eventually let `make pull_translations` pull from https://github.com/openedx/openedx-translations instead of Transifex, which has been concluded by [OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

This has been generated with the following command:

```
$ cd edx-platform
$ make upgrade-package package=openedx-atlas
```

Which is inspired by the [upgrade-one-python-dependency.yml](https://github.com/openedx/edx-platform/blob/master/.github/workflows/upgrade-one-python-dependency.yml) workflow which I suppose I cannot run due to the lack of GitHub permission.


## Testing instructions

 - [x] Build the image on my fork
 - [x] Upgrade to `openedx-atlas==0.5.0` 
 - [x] Test again

```
$ DOCKER_BUILDKIT=1 docker build . --build-arg SERVICE_VARIANT=lms --build-arg SERVICE_PORT=8000 --target development -t openedx/lms-dev
...
$ docker run -it docker.io/openedx/lms-dev atlas --version
v0.5.0

...

devstack $ make dev.up.lms
devstack $ make dev.shell.lms
root@lms:/edx/app/edxapp/edx-platform# atlas --version
v0.5.0


```


 - [x] Run atlas and pull translations on `edx-platform`


<details><summary>Testing atlas pull with --expand-glob</summary>

```
$ docker run -it docker.io/openedx/lms-dev bash
root@c8b826df60ce:/edx/app/edxapp/edx-platform# rm -rf conf/plugins-locale
root@c8b826df60ce:/edx/app/edxapp/edx-platform# atlas pull --expand-glob 'translations/*/done/conf/locale:conf/plugins-locale/done'
Pulling translation files
 - directory: translations/*/done/conf/locale:conf/plugins-locale/done
 - repository: openedx/openedx-translations
 - branch: main
 - filter: Not Specified
 - expand-glob: 1
Creating a temporary Git repository to pull translations into "./translations_TEMP"...
Done.
Setting git sparse-checkout rules...
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 4 (delta 3), reused 1 (delta 1), pack-reused 0
Receiving objects: 100% (4/4), 2.78 KiB | 2.78 MiB/s, done.
Resolving deltas: 100% (3/3), done.
Done.
Pulling translation files from the repository...
Your branch is up to date with 'origin/main'.
Done.
Copying translations from "./translations_TEMP/translations/DoneXBlock/done/conf/locale" to "./conf/plugins-locale/done"...
Done.
Removing temporary directory...
Done.

Translations pulled successfully!
root@c8b826df60ce:/edx/app/edxapp/edx-platform# find conf/plugins-locale/ -type f
conf/plugins-locale/done/fr_CA/LC_MESSAGES/django.po
conf/plugins-locale/done/en/LC_MESSAGES/django.po
conf/plugins-locale/done/de/LC_MESSAGES/django.po
conf/plugins-locale/done/ar/LC_MESSAGES/django.po

```


</details> 


## Supporting information

This contribution is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

Up-to-date project overview and details are available in the [Approach Memo and Technical Discovery: Translations Infrastructure Implementation](https://docs.google.com/document/d/11dFBCnbdHiCEdZp3pZeHdeH8m7Glla-XbIin7cnIOzU/edit#) document.

Join the conversation on [Open edX Slack #translations-project-fc-0012](https://openedx.slack.com/archives/C04R6TUJB7T).

Check the links above for full information about the overall project.

Internalization is being rearchitected in Open edX Python, XBlock, Micro-frontend, and other projects. There are a number of immediately visible changes:
 - Remove source and language translations from the repositories, hence no `.json`, `.po` or `.mo` files will be committed into the repos.
 - Add standardized `make extract_translations` in all repositories
 - Push user-facing messages strings into [openedx/openedx-translations](https://github.com/openedx/openedx-translations/).
 - Integrate root repositories with [openedx/openedx-atlas](https://github.com/openedx/openedx-atlas/) to pull translations on build/deploy time

openedx-webhooks commented:
Thanks for the pull request, @OmarIthawi!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

brian-smith-tcril commented:
I think we should hold off on this until we implement https://github.com/openedx/openedx-atlas/issues/34

OmarIthawi committed:
feat: install `atlas` translation tool

Add the openedx-atlas to the kernel.in so it's available in all
environments including Atlas, Devstack and Native installation.

OmarIthawi commented:
> I think we should hold off on this until we implement [openedx/openedx-atlas#34](https://github.com/openedx/openedx-atlas/issues/34)

@brian-smith-tcril This is now ready for another review.

brian-smith-tcril commented:
@openedx/teaching-and-learning you're listed as "Review required from:" for this repo on the OEP-58/FC-12 repo reviewers spreadsheet. Would you mind taking a look at this one?

brian-smith-tcril commented:
@jristau1984 @jmbowman I'm tagging you because you're listed as "Review required from:" for this repo on the FC-0012/OEP-58 spreadsheet

robrap reviewed (approved):
Approving on behalf of @jmbowman's team. I'm looking forward to translations being handled outside of Jenkins. Thank you.

robrap commented:
@OmarIthawi @brian-smith-tcril: I know I just approved this, but I realize I do have a question. Is atlas actually needed in Production? Does it provide a potential security risk being a production dependency, rather than using another requirement file that can be used only where it is needed?

OmarIthawi commented:
> @OmarIthawi @brian-smith-tcril: I know I just approved this, but I realize I do have a question. Is atlas actually needed in Production? Does it provide a potential security risk being a production dependency, rather than using another requirement file that can be used only where it is needed?

@robrap Good question. My thinking is it's safe to have in production.

[Atlas](https://github.com/openedx/openedx-atlas/blob/dd55d8921cf3952be3b9a962139184083dd9e88c/atlas) is a bash script that is calls `git`, and yes it's needed for production environments like Native Installation.

It has no server component, requirements or even a functional Python module. It's published on PyPi to make use of `pip-tools` and other dependency management tools. 

Docker-based environments may benefit from deleting it but there's a very small benefit from doing so.

> Approving on behalf of @jmbowman's team. I'm looking forward to translations being handled outside of Jenkins. Thank you.

You're welcome 😃

OmarIthawi commented:
@brian-smith-tcril @robrap would you mind merging this pull request? It's been tested and reviewed.

cc: @mphilbrick211

openedx-webhooks commented:
@OmarIthawi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/32952&entry.1484655318=https://github.com/OmarIthawi&entry.752974735=2023-08-09+12:41&entry.1917517419=2023-09-13+12:11&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
